File: rust/src/bin/2015-03.rs
use std::collections::HashSet;
use my_project::get_input;

fn received_presents_count(content: &str, deliverers_count: usize) -> usize{
    let mut start_locs = vec![(0, 0); deliverers_count]; // Deliverers positions
    let mut visited_houses = HashSet::new(); // Houses that got at least one present
    visited_houses.insert(start_locs[0]);

    for (id, char) in content.bytes().enumerate() {
        let deliverer_id = id % start_locs.len();
        match char {
            b'^' => start_locs[deliverer_id].0 -= 1,
            b'v' => start_locs[deliverer_id].0 += 1,
            b'<' => start_locs[deliverer_id].1 -= 1,
            b'>' => start_locs[deliverer_id].1 += 1,
            _ => {}
        }
        visited_houses.insert(start_locs[deliverer_id]);
    }
    visited_houses.len()
}

fn main() {
    let content = get_input(2015, 3).expect("Can't read from file");

    println!("Part A: {}", received_presents_count(&content, 1));
    println!("Part B: {}", received_presents_count(&content, 2));
}

File: python/2015-03.py
from inputGetter import get_input

content = get_input(2015, 3)

start_loc = [0, 0]
res_a = set()
res_a.add(tuple(start_loc))

for char in content:
    if char == '^':
        start_loc[0] -= 1
    elif char == 'v':
        start_loc[0] += 1
    elif char == '<':
        start_loc[1] -= 1
    else:
        start_loc[1] += 1
    res_a.add(tuple(start_loc))

print(f'Part A: {len(res_a)}')

start_loc = [[0, 0], [0, 0]]
res_b = [set(), set()]
res_b[0].add(tuple(start_loc[0]))

for i in range(0, len(content), 2):
    for j in range(2):
        char = content[i+j]
        if char == '^':
            start_loc[j][0] -= 1
        elif char == 'v':
            start_loc[j][0] += 1
        elif char == '<':
            start_loc[j][1] -= 1
        else:
            start_loc[j][1] += 1
        res_b[j].add(tuple(start_loc[j]))

print(f'Part B: {len(res_b[0] | res_b[1])}')